wasm/jit/compiler.go

```diff
@@ -8,86 +8,311 @@ import (
 // compiler is the interface of architecture-specific native code compiler,
 // and this is responsible for compiling native code for all wazeroir operations.
 type compiler interface {
-	// For debugging purpose.
+	// String is for debugging purpose.
 	String() string
 	// emitPreamble is called before compiling any wazeroir operation.
 	// This is used, for example, to initilize the reserved registers, etc.
 	emitPreamble() error
-	// Generates the byte slice of native codes.
+	// generate generates the byte slice of native code.
 	// maxStackPointer is the max stack pointer that the target function would reach.
+	// staticData is compiledFunctionStaticData for the resutling native code.
 	generate() (code []byte, staticData compiledFunctionStaticData, maxStackPointer uint64, err error)
-	// Emit the trampoline code from which native code can jump into the host function.
+	// compileHostFunction emits the trampoline code from which native code can jump into the host function.
 	// TODO: maybe we wouldn't need to have trampoline for host functions.
 	compileHostFunction(address wasm.FunctionAddress) error
+	// compileLabel notify compilers of the beginning of a label.
 	// Return true if the compiler decided to skip the entire label.
 	compileLabel(o *wazeroir.OperationLabel) (skipThisLabel bool)
-	// Followings are resinposible for compiling each wazeroir operation.
+	// compileUnreachable adds instructions to return to engine with jitCallStatusCodeUnreachable status.
+	// See wasm.OpcodeUnreachable.
 	compileUnreachable() error
+	// compileSwap adds instruction to swap the stack top value with the target in the Wasm value stack.
+	// The values are might be on registers or memory-stack at runtime, so compiler implementations
+	// emit instructions to swap values depending these locations.
 	compileSwap(o *wazeroir.OperationSwap) error
+	// compileGlobalGet adds instructions to read the value of the given index in the ModuleInstance.Globals
+	// and push the value onto the stack.
+	// See wasm.OpcodeGlobalGet.
 	compileGlobalGet(o *wazeroir.OperationGlobalGet) error
+	// compileGlobalSet adds instructions to set the top value on the stack to the given index in the ModuleInstance.Globals.
+	// See wasm.OpcodeGlobalSet.
```

Comment:
no need to trailing dot a code link and it may confuse things (here and everywhere)